tests/basic/pure_test.go

```diff
@@ -1865,3 +1865,155 @@ var _ = Describe("{CreateAndDeleteMultipleVolumesInParallel}", func() {
+			}
+		}
+
+		Step("Schedule applications", func() {
+			log.InfoD("Scheduling applications")
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+				taskName := fmt.Sprintf("clone-validate-%d", i)
+				for _, ctx := range ScheduleApplications(taskName) {
+					ctx.ReadinessTimeout = appReadinessTimeout
+					contexts = append(contexts, ctx)
+				}
+			}
+		})
+		Step("Validate applications", func() {
+			log.InfoD("Validating applications")
+			ValidateApplications(contexts)
+		})
+		Step("Identify backend and categorize volumes", func() {
+			log.InfoD("Identifying backend")
+			volDriverNamespace, err := Inst().V.GetVolumeDriverNamespace()
+			log.FailOnError(err, "failed to get volume driver [%s] namespace", Inst().V.String())
+			secretList, err := core.Instance().ListSecret(volDriverNamespace, metav1.ListOptions{})
+			log.FailOnError(err, "failed to get secret list from namespace [%s]", volDriverNamespace)
+			for _, secret := range secretList.Items {
+				switch secret.Name {
+				case PX_PURE_SECRET_NAME:
+					backend = BackendPure
+					break
+				case PX_VSPHERE_SCERET_NAME:
+					backend = BackendVSphere
+					break
+				}
+			}
+			log.InfoD("Backend: %v", backend)
+			log.InfoD("Categorizing volumes")
+			for _, ctx := range contexts {
+				volumes, err := Inst().S.GetVolumes(ctx)
+				log.FailOnError(err, "failed to get volumes for app [%s/%s]", ctx.App.NameSpace, ctx.App.Key)
+				dash.VerifyFatal(len(volumes) > 0, true, "Verifying if volumes exist for creating clone")
+				// The CloudStorage.Provider in StorageCluster Spec is not accurate
+				for _, vol := range volumes {
+					apiVol, err := Inst().V.InspectVolume(vol.ID)
+					log.FailOnError(err, "failed to inspect volume [%s/%s]", vol.Name, vol.ID)
+					switch backend {
+					case BackendPure:
+						volType, err := getPureVolumeType(vol)
+						log.FailOnError(err, "failed to get pure volume type for volume [%+v]", vol)
+						volumeMap[volType] = append(volumeMap[volType], apiVol)
+					case BackendVSphere:
+						volumeMap[VolumeVsCD] = append(volumeMap[VolumeVsCD], apiVol)
+					default:
+						volumeMap[VolumeUnknown] = append(volumeMap[VolumeUnknown], apiVol)
+					}
+				}
+			}
+		})
+		Step(fmt.Sprintf("Create a clone for each volume"), func() {
+			log.InfoD("Create a clone for each volume")
+			for volType, vols := range volumeMap {
+				for _, vol := range vols {
+					cloneName := fmt.Sprintf("%s-clone", vol.Locator.Name)
+					log.Infof("Create clone [%s] for [%s] volume [%s/%s]", cloneName, volType, vol.Id, vol.Locator.Name)
+					clone, err := Inst().V.CloneVolume(vol.Id)
+					log.FailOnError(err, "failed to create clone  for [%s] volume [%s/%s]", volType, vol.Id, vol.Locator.Name)
+					cloneInspect, err := Inst().V.InspectVolume(clone)
+					log.FailOnError(err, "failed to inspect clone [%s]  for [%s] volume [%s/%s]", cloneInspect.Id, volType, vol.Id, vol.Locator.Name)
+					volumeCloneMap[volType] = append(volumeCloneMap[volType], cloneInspect)
```

Comment:
can you explain how this categorisation is used?

Reply:
This volcloneMap is used to store clone of various volType just as volumeMap.